domain/consensus/model/externalapi/consensus.go

```diff
@@ -40,7 +40,9 @@ type Consensus interface {
 	PopulateMass(transaction *DomainTransaction)
 }
 
-// ConsensusWrapper is a wrapper for consensus
-type ConsensusWrapper interface {
+// ConsensusReference holds a reference for consensus.
+// It is used when you want access to a consensus that
+// can be swapped.
```

Comment:
Please consider:
ConsensusReference holds a reference to a consensus object.
The consensus object may be swapped with a new one entirely during the IBD process. Before an atomic consensus operation, callers are expected to call Consensus() once and work against that instance throughout.

Reply:
Also, this isn't strictly a consensus interface. Please consider moving it into somewhere in `domain`